### PR TITLE
ci(proto): add proto formatting, cleanup justfile

### PR DESCRIPTION
**Pull request by joroshiba**
## Summary
Cleans up some justfile, updates to check protos against buf, updates readmes,  and adds default formatting of proto files. 

## Background
We moved protos, this caused an issue with breaking changes. Additionally we had non consistent proto formatting.

## Changes
- remove no longer needed just commands
- restructure fmt & lint to take args, add an `all` which is default
- add buf format, run against the repo
- use the buf posted files for breaking changes checks.
- update readmes

## Testing
CI/CD tests, manual testing commands

### CODE REVIEW

File: justfile [hunk 1]
```diff
@@ -1,46 +1,67 @@
...
+  @just _fmt-rust 
+  @just _fmt-toml
+  @just _fmt-proto
 
-fmt-rust:
+@_lint-all: 
+  -just _lint-rust 
+  -just _lint-toml 
+  -just _lint-proto
+  -just _lint-md
```
Comment (joroshiba) [inline justfile:37]: Error codes are swallowed on lint so all outputs will be listed

File: justfile [hunk 2]
```diff
@@ -1,46 +1,67 @@
...
+# Can lint 'rust', 'toml', 'proto', 'md' or 'all'. Defaults to all.
+lint lang=default_lang:
+  @just _lint-{{lang}}
 
-docker-build crate tag=default_docker_tag:
-  docker buildx build --load --build-arg TARGETBINARY={{crate}} -f containerfiles/Dockerfile -t {{crate}}:{{tag}} .
+_fmt-all: 
+  @just _fmt-rust 
+  @just _fmt-toml
+  @just _fmt-proto
```
Comment (joroshiba) [inline justfile:31]: this command is run silently, since it's just a pass through

Comment (SuperFluffy) [reply justfile:31]: also very cool

File: justfile (hunk 1 above)
Comment (SuperFluffy) [reply justfile:37]: I am not sure what this means.

**Review by SuperFluffy — APPROVED** [review]
These are great. I didn't know about all these just features.

**Comment by joroshiba** [comment]
The output from `just` or `just --list` now:

```bash
Available recipes:
    compile-protos         # Compiles the generated rust code from protos which are used in crates.
    default
    docker-build crate tag=default_docker_tag # Builds docker image for the crate. Defaults to 'local' tag.
    fmt lang=default_lang  # Can format 'rust', 'toml', 'proto', or 'all'. Defaults to all
    install-cli
    lint lang=default_lang # Can lint 'rust', 'toml', 'proto', 'md' or 'all'. Defaults to all.
```

**Review by noot — APPROVED** [review]

**Pull request merged**